service/crt/service.go

```diff
@@ -194,8 +193,9 @@ func (s *Service) newCertificateListWatch() *cache.ListWatch {
 				return nil, err
 			}
 
-			watcher := watch.NewStreamWatcher(&k8sutil.CertificateDecoder{
-				Stream: stream,
+			watcher := watch.NewStreamWatcher(&certificateDecoder{
+				decoder: json.NewDecoder(stream),
+				close:   stream.Close,
```

Comment:
The decoder here and in `aws-operator` is missing this close func which is definitely needed. To get it working I just moved the decoder into the service.

@xh3b4sd this was based on the decoder in `kvm-operator` which I love <3 Its also in `operatorkit` so we can fix it by using the code there.